src/main/php/com/handlebarsjs/PartialBlockHelper.class.php

```diff
@@ -0,0 +1,46 @@
+<?php namespace com\handlebarsjs;
+
+use com\github\mustache\NodeList;
+use com\github\mustache\TemplateNotFoundException;
+
+/**
+ * Partial blocks
+ *
+ * @see   http://handlebarsjs.com/partials.html
+ * @test  xp://com.handlebarsjs.unittest.PartialBlockHelperTest
+ */
+class PartialBlockHelper extends BlockNode {
+
+  /**
+   * Creates a new with block helper
+   *
+   * @param string[] $options
+   * @param com.github.mustache.NodeList $fn
+   * @param com.github.mustache.NodeList $inverse
+   * @param string $start
+   * @param string $end
+   */
+  public function __construct($options= [], NodeList $fn= null, NodeList $inverse= null, $start= '{{', $end= '}}') {
+    $template= (string)array_shift($options);
+    parent::__construct($template, $options, $fn, $inverse, $start, $end);
+  }
+
+  /**
+   * Evaluates this node
+   *
+   * @param  com.github.mustache.Context $context the rendering context
+   * @return string
+   */
+  public function evaluate($context) {
+    $templates= $context->engine->getTemplates();
+
+    try {
+      $templates->register('@partial-block', $this->fn);
+      return $context->engine->transform($this->name, $context, $this->start, $this->end, '');
+    } catch (TemplateNotFoundException $e) {
```

Comment:
It would be nicer if mustache template loaders would either:

* have a `provides()` method to test for template existance, *or*
* accept a `default` closure to execute if the template does not exist

Catching an exception for control flow does not feel right [TM]

Reply:
See https://github.com/xp-forge/mustache/pull/8